projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -2090,14 +2181,16 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
                 }
                 const maxSize = Math.ceil(Math.max(...contentWidths));
                 dim.autoWidth = maxSize;
+            } else if (dim.width === 'auto') {
+                this.autoSizeRowDimension(dim);
```

Comment:
This still needs a check whether the cells have rendered before attempting to measure them. Currently setting width:"auto" with horizontal mode, doesn't seem to work since the measuring happens before the cells have rendered.

Reply:
Yeah forgot about that. I changed it a bit since it wasn't using initial auto size like the vertical. There's still an issue when row dimension headers are enabled. For some reason `getNodeSizeViaRange` returns different values initially and when double clicking on header resize. I'll log that separately since this is not related and I can seem to find a quick solution except `setTimeout` which causes the grid to flicker initially.